File: 00. Estructura/01. Lenguajes de proposito general/02. Programación básica/07. Funciones/06. Con paramatros y con retorno multiple/JavaScript/Con paramatros y con retorno multiple.js

// Creación con parametros limitados
function funcion(var1, var2, varN) {
    // Contenido
    return [valor1, valor2, valorN];
}

// Creación con parametros ilimitados
function funcion(...variables) {
    // Contenido
    return [valor1, valor2, valorN];
}

// Creación con parametros por defecto (Los valores por defecto de pueden omitir en el llamado)
function funcion(var1 = val1, var2 = val2, varN = valN) {
    // Contenido
    return [valor1, valor2, valorN];
}
 
// Llamado
/*Contexto*/ funcion(valor1, valor2, valorN) /*Contexto*/

// Llamado con asignación
variables = funcion(valor1, valor2, valorN);
varible1 = variables[0];
varible2 = variables[1];
varibleN = variables[2];

// Eliminación
NO SOPORTADO (EL GARBAGE COLLECTOR LA ELIMINA AUTOMATICAMENTE CUANDO YA ES INACCESIBLE)

File: 00. Estructura/01. Lenguajes de proposito general/02. Programación básica/07. Funciones/06. Con paramatros y con retorno multiple/MathLab/Con paramatros y con retorno multiple.m

% Llamado
'Contexto' funcion(valor1, valor2, valorN) 'Contexto'

% Llamado en asignación
[var1,var2, varN] = funcion(valor1, valor2, valorN)

% Creación con parametros limitados
function [ret1,ret2, retN] = funcion(var1, var2, varN)
    % Contenido
    ret1 = val1;
    ret2 = val2;
    retN = valN;
end

% Creación con parametros ilimitados
function [ret1,ret2, retN] = funcion(varargin)
    % Contenido
    'Contexto' varargin{3} 'Contexto'    % Obtener parametro en la posición especificada
    % Contenido
    ret1 = val1;
    ret2 = val2;
    retN = valN; 
end

% Eliminación
NO SOPORTADO
